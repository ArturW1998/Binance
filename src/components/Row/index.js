import React from 'react';
import PropTypes from 'prop-types';
import BigNumber from 'bignumber.js';

const Row = ({
  symbol,
  latestPrice,
  highPrice,
  lowPrice,
  quoteVolume,
  openPrice,
}) => (
  <li className="table-item">
    <article className="d-none d-md-inline">
      <div className="row table-row small py-1">
        <div className="col">{symbol}</div>
        <div className="col">
          {new BigNumber(latestPrice).toFormat(null, 1)}
        </div>
        <div className="col">{new BigNumber(openPrice).toFormat(null, 1)}</div>
        <div className="col">{new BigNumber(highPrice).toFormat(null, 1)}</div>
        <div className="col">{new BigNumber(lowPrice).toFormat(null, 1)}</div>
        <div className="col">
          {new BigNumber(quoteVolume).toFormat(null, 1)}
        </div>
      </div>
    </article>

    <article className="d-inline d-md-none">
      <div className="row table-row small py-1">
        <div className="col-4">
          <div className="font-weight-light text-muted small">Pair</div>
          <span className="font-weight-bold">{symbol}</span>
        </div>
        <div className="col-4">
          <div className="font-weight-light text-muted small">
            Lastest Price
          </div>
          <span>{new BigNumber(latestPrice).toFormat(null, 1)}</span>
        </div>
        <div className="col-4">
          <div className="font-weight-light text-muted small">Open</div>
          <span className="small">
            {new BigNumber(openPrice).toFormat(null, 1)}
          </span>
        </div>
        <div className="col-4">
          <div className="font-weight-light text-muted small">High</div>
          <span className="small">
            {new BigNumber(highPrice).toFormat(null, 1)}
          </span>
        </div>
        <div className="col-4">
          <div className="font-weight-light text-muted small">Low</div>
          <span className="small">
            {new BigNumber(lowPrice).toFormat(null, 1)}
          </span>
        </div>
        <div className="col-4">
          <div className="font-weight-light text-muted small">Volume</div>
          <span className="small">
            {new BigNumber(quoteVolume).toFormat(null, 1)}
          </span>
        </div>
      </div>
    </article>
  </li>
);

Row.propTypes = {
  symbol: PropTypes.string,
  latestPrice: PropTypes.string,
  highPrice: PropTypes.string,
  lowPrice: PropTypes.string,
  quoteVolume: PropTypes.string,
  openPrice: PropTypes.string,
};

Row.defaultProps = {
  symbol: '',
  latestPrice: '',
  highPrice: '',
  lowPrice: '',
  quoteVolume: '',
  openPrice: '',
};

export default Row;
