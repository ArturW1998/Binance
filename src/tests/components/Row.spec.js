import React from 'react';
import { shallow } from 'enzyme';
import Row from '../../components/Row';

describe('Row component', () => {
  const props = {
    symbol: 'BNBBTC',
    latestPrice: '0.00184990',
    openPrice: '0.00187580',
    highPrice: '0.00182500',
    lowPrice: '2332.46079005',
    quoteVolume: '0.00184780',
  };

  const row = shallow(<Row {...props} />);

  it('renders properly', () => {
    expect(row.html()).toMatchSnapshot();
  });

  it('renders table item', () => {
    expect(row.find('.table-item')).toHaveLength(1);
  });

  it('renders symbol', () => {
    expect(row.find('.title').text()).toEqual(props.symbol);
  });

  describe('correct render columns', () => {
    it('renders 6 middle columns', () => {
      expect(row.find('.col')).toHaveLength(6);
    });

    it('renders 6 small columns', () => {
      expect(row.find('.col-4')).toHaveLength(6);
    });
  });

  describe('correct render for first row', () => {
    it('Pair', () => {
      expect(
        row
          .find('h2')
          .first()
          .text(),
      ).toEqual('Pair');
    });

    it('Latest Price', () => {
      expect(
        row
          .find('h2')
          .at(1)
          .text(),
      ).toEqual('Latest Price');
    });
  });
});
