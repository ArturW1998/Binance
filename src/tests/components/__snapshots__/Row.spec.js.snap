// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component renders properly 1`] = `"<li class=\\"table-item\\"><article class=\\"d-none d-md-inline\\"><div class=\\"row table-row small py-1\\"><div class=\\"col d-flex align-items-center\\"><h3 class=\\"title font-weight-normal mb-0\\">BNBBTC</h3></div><div class=\\"col d-flex align-items-center\\"><span>0.00184990</span></div><div class=\\"col d-flex align-items-center\\"><span>0.00182500</span></div><div class=\\"col d-flex align-items-center\\"><span>2332.46079005</span></div><div class=\\"col d-flex align-items-center\\"><span>0.00184780</span></div><div class=\\"col d-flex align-items-center\\"><span>0.00187580</span></div></div></article><article class=\\"d-inline d-md-none\\"><div class=\\"row table-row small py-1\\"><div class=\\"col-4\\"><h2 class=\\"font-weight-light text-muted small mb-0\\">Pair</h2><span class=\\"font-weight-bold\\">BNBBTC</span></div><div class=\\"col-4\\"><h2 class=\\"font-weight-light text-muted small mb-0\\">Latest Price</h2><span class=\\"small\\">0.00184990</span></div><div class=\\"col-4\\"><h2 class=\\"font-weight-light text-muted small mb-0\\">Open</h2><span class=\\"small\\">0.00182500</span></div><div class=\\"col-4\\"><h2 class=\\"font-weight-light text-muted small mb-0\\">High</h2><span class=\\"small\\">2332.46079005</span></div><div class=\\"col-4\\"><h2 class=\\"font-weight-light text-muted small mb-0\\">Low</h2><span class=\\"small\\">0.00184780</span></div><div class=\\"col-4\\"><h2 class=\\"font-weight-light text-muted small mb-0\\">Volume</h2><span class=\\"small\\">0.00187580</span></div></div></article></li>"`;
